grust/templates/sys/crate.tmpl

```diff
@@ -206,11 +207,15 @@ pub struct ${node.ctype}(gpointer);
 pub enum ${node.ctype} { }
 %   else:
 #[repr(C)]
+%       if not mapper.struct_is_valid(node):
+pub struct ${node.ctype};
```

Comment:
This might work for objects and some boxed types, but I wouldn't like to get unusable types for copyable values. What I did in the few crate-specific templates I created in repo `gi-rust/crates` is suppress the unmappable structures and place manual representations in the crate template. With support for unions available in stable Rust, there will be less need for these overrides.

Bit fields could be automated too, if there is a equivalent `#[repr(C)]` guaranteed by the C standard.